Integrations/python/deephaven/Types.py

```diff
@@ -0,0 +1,311 @@
+    return jclass
+
+
+@_passThrough
+def _jpyTypeFromType(data_type : DataType):
+    if data_type is None:
+        return None
+    type2jtype = {
+        bool_ : jpy.get_type('java.lang.Boolean'),
+        byte : jpy.get_type('byte'),
+        short : jpy.get_type('short'),
+        int_ : jpy.get_type('int'),
+        long_ : jpy.get_type('long'),
+        float_ : jpy.get_type('float'),
+        double : jpy.get_type('double'),
+        string : jpy.get_type('java.lang.String'),
+        bigdecimal : jpy.get_type('java.math.BigDecimal'),
+        stringset : jpy.get_type('io.deephaven.db.tables.libs.StringSet'),
+        datetime : jpy.get_type('io.deephaven.db.tables.utils.DBDateTime'),
+        byte_array : jpy.get_type('[B'),
+        short_array : jpy.get_type('[S'),
+        int_array : jpy.get_type('[I'),
+        long_array : jpy.get_type('[J'),
+        float_array : jpy.get_type('[S'),
+        double_array : jpy.get_type('[D'),
+        string_array : jpy.get_type('[Ljava.lang.String;')
+    }
+    jpy_type = type2jtype.get(data_type, None)
+    if jpy_type is not None:
+        return jpy_type
+    jclass = _jclassFromType(data_type)
+    return jpy_get_type(jclass.getName())
+
+
+@_passThrough
+def _isPrimitive(data_type : DataType):
+    primitives = { bool_, byte, short, int_, long_, float_, double }
+    return data_type in primitives
+
+@_passThrough
+def _colDef(col_name : str, data_type : DataType, component_type : DataType = None):
+    """
+    Create a ColumnDefinition object.
+    :param col_name: The column's new.
+    :param data_type: The column's data type.
+    :param component_type: The column's component type, or None if none.
+    :return: the column definition object.
+    """
+    jdata_type = _jclassFromType(data_type)
+    jcomponent_type = _jclassFromType(component_type)
+
+    return _col_def_.fromGenericType(col_name, jdata_type, jcomponent_type)
+
+@_passThrough
+def _colDefs(ts):
+    """
+    Convert a sequence of tuples of the form ('Price', double_type)
+    or ('Prices', double_array_type, double_type)
+    to a list of ColumnDefinition objects.
+
+    :param ts: a sequence of 2 or 3 element tuples of (str, type) or (str, type, type)  specifying a column definition object.
+    :return: a list of column definition objects.
+    """
+    r = []
+    for t in ts:
+        r.append(_colDef(*t))
+    return r
+
+
+@_passThrough
+def _getQstCol(col_name:str, col_type:DataType, col_data=None):
+    if col_data is None or len(col_data) < 1:
+        col_header = _qst_col_header_.of(col_name, col_type)
+        return _qst_column_.empty(col_header)
+    jtype = _jpyTypeFromType(col_type)
+    if jtype is None:
+        raise Exception("value for argument 'col_type' " +
+                        str(col_type) + " is not a known data type.")
+    jvalues = jpy.array(jtype, col_data)
+    if _isPrimitive(col_type):
+        return _qst_column_.ofUnsafe(col_name, jvalues)
```

Comment:
This will break for `bool_`, as written.

Reply:
Fixed.